src/mem3/test/eunit/mem3_distribution_test.erl

```diff
@@ -0,0 +1,62 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(mem3_distribution_test).
+
+-include_lib("couch/include/couch_eunit.hrl").
+
+-define(TDEF_FE(Name), fun(Arg) -> {atom_to_list(Name), ?_test(Name(Arg))} end).
+-define(MOD, mem3_distribution).
+
+setup() ->
+    Ctx = test_util:start_couch([mem3]),
+    meck:new(mem3, [passthrough]),
+    meck:new(mem3_distribution, [passthrough]),
+    Ctx.
+
+teardown(Ctx) ->
+    meck:unload(),
+    test_util:stop_couch(Ctx).
+
+mem3_distribution_test_() ->
+    {
+        foreach,
+        fun setup/0,
+        fun teardown/1,
+        [
+            ?TDEF_FE(periodic_scheduler_works),
+            ?TDEF_FE(connect_to_unconnected_nodes)
+        ]
+    }.
+
+periodic_scheduler_works(_) ->
+    St = sys:get_state(?MOD),
+    {st, TRef} = St,
+    TVal = erlang:read_timer(TRef),
+    ?assert(is_integer(TVal)),
+    ?assert(TVal > 0),
+    ?assert(TVal =< 70000),
+    {noreply, St1} = ?MOD:handle_info(connect, St),
+    {st, TRef1} = St1,
+    ?assertNotEqual(TRef, TRef1),
+    TVal1 = erlang:read_timer(TRef1),
+    ?assert(is_integer(TVal1)),
+    ?assertNotEqual(TVal, TVal1).
+
+connect_to_unconnected_nodes(_) ->
+    Nodes = ['foo', 'bar'],
+    meck:expect(mem3, nodes, 0, Nodes),
+    meck:reset(?MOD),
+    % Simulate connect timer expiry
+    ?MOD ! connect,
+    meck:wait(?MOD, connect_node, [foo], 5000),
+    meck:wait(?MOD, connect_node, [bar], 5000).
```

Comment:
Do you think it's worth also checking whether these result in "reconnected" log messages? Maybe I'm misunderstanding the code, but I was expecting this test to cause two warning "reconnected" log entries, but I don't see them in `.eunit/couch.log`, nor does coverage indicate they ran:
<img width="814" alt="Screen Shot 2022-09-21 at 12 53 34 PM" src="https://user-images.githubusercontent.com/51209/191599149-f38d3cf9-6ea6-4f78-b58b-2ff22ab84b67.png">
Perhaps the test finishes before the logging would happen?

Ah, never mind, I see that it's only expected to log if `connect_node` returns true, which it would not for made up `foo` and `bar` nodes, unless it was also mocked.

Reply:
A good idea, I'll make it `true` and `false`